apps/emqx_rule_engine/src/emqx_rule_runtime.erl

```diff
@@ -139,25 +139,35 @@ apply_rule(Rule = #{id := RuleID}, Columns, Envs) ->
         reset_process_trace_metadata(Columns)
     end.
 
-set_process_trace_metadata(RuleID, #{clientid := ClientID}) ->
+set_process_trace_metadata(RuleID, #{clientid := ClientID} = Columns) ->
     logger:update_process_metadata(#{
-        rule_id => RuleID,
         clientid => ClientID
-    });
-set_process_trace_metadata(RuleID, _) ->
+    }),
+    set_process_trace_metadata(RuleID, maps:remove(clientid, Columns));
```

Comment:
nit: do not build a new map, instead, add a function like `add_rule_trigger_time(#{clientid => ClientID, rule_id => RuleID})`

Reply:
Ok, thanks, I will try to fix this in another PR.

Reply:
Added a commit to PR https://github.com/emqx/emqx/pull/12916 that simplifies the function:



See [841afbc](https://github.com/emqx/emqx/pull/12916/commits/841afbc1404d7ea53acb4a7a7d8a0ee0201e240d)